token/token_blackbox_test.go

```diff
@@ -20,125 +22,129 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestServiceAccount(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
+type TokenManagerTestSuite struct {
+	testsuite.UnitTestSuite
+	tm     token.Manager
+	config *tokensupport.ManagerConfigurationMock
+}
+
+func TestRunTokenManagerTestSuite(t *testing.T) {
+	suite.Run(t, &TokenManagerTestSuite{UnitTestSuite: testsuite.NewUnitTestSuite()})
+}
+
+func (s *TokenManagerTestSuite) SetupSuite() {
+	s.UnitTestSuite.SetupSuite()
 
+	s.config = testconfiguration.NewDefaultMockTokenManagerConfiguration(s.T())
+	var err error
+	s.tm, err = token.NewManager(s.config)
+	require.NoError(s.T(), err)
+}
+
+func (s *TokenManagerTestSuite) TestServiceAccount() {
 	serviceName := "test-service"
 
-	t.Run("Valid", func(t *testing.T) {
+	s.T().Run("Valid", func(t *testing.T) {
 		// given
 		claims := jwt.MapClaims{}
 		claims["service_accountname"] = serviceName
 		ctx := goajwt.WithJWT(context.Background(), jwt.NewWithClaims(jwt.SigningMethodRS512, claims))
 		// then
-		assert.True(t, token.IsServiceAccount(ctx))
+		assert.True(s.T(), token.IsServiceAccount(ctx))
 	})
-	t.Run("Missing name", func(t *testing.T) {
+	s.T().Run("Missing name", func(t *testing.T) {
 		// given
 		claims := jwt.MapClaims{}
 		ctx := goajwt.WithJWT(context.Background(), jwt.NewWithClaims(jwt.SigningMethodRS512, claims))
 		// then
-		assert.False(t, token.IsServiceAccount(ctx))
+		assert.False(s.T(), token.IsServiceAccount(ctx))
 	})
-	t.Run("Missing token", func(t *testing.T) {
+	s.T().Run("Missing token", func(t *testing.T) {
 		// given
 		ctx := context.Background()
 		// then
-		assert.False(t, token.IsServiceAccount(ctx))
+		assert.False(s.T(), token.IsServiceAccount(ctx))
 	})
-	t.Run("Nil token", func(t *testing.T) {
+	s.T().Run("Nil token", func(t *testing.T) {
 		// given
 		ctx := goajwt.WithJWT(context.Background(), nil)
 		// then
-		assert.False(t, token.IsServiceAccount(ctx))
+		assert.False(s.T(), token.IsServiceAccount(ctx))
 	})
-	t.Run("Wrong data type", func(t *testing.T) {
+	s.T().Run("Wrong data type", func(t *testing.T) {
 		// given
 		claims := jwt.MapClaims{}
 		claims["service_accountname"] = 100
 		ctx := goajwt.WithJWT(context.Background(), jwt.NewWithClaims(jwt.SigningMethodRS512, claims))
 		// then
-		assert.False(t, token.IsServiceAccount(ctx))
+		assert.False(s.T(), token.IsServiceAccount(ctx))
```

Comment:
No, this has to be `t` and not `s.T()` here otherwise in case of assertion failure, the test will panic. The current test is `t`, and `s.T()` is the "parent" test.

Reply:
This change needs to be reverted everywhere else in this file.

Reply:
It's a pointer to the same `t`. There is no parent/child `testing.T` or I'm missing something?
But anyway. It was a bulk update and I now reverted `s.T()` to `t` in subtests because the code looks cleaner with that - https://github.com/fabric8-services/fabric8-common/pull/25/commits/8a492bef96d08c8927399914301232df2b91f4f4
But I still don't see any issue with using s.T() here.

Reply:
OK. My bad. I looked at the Run() but missed the t = &T{} part somehow.. Now I see it. Thanks.